Source/SvgElement.cs

```diff
@@ -1042,7 +1042,18 @@ public virtual SvgElement DeepCopy<T>() where T : SvgElement, new()
                 newObj.CustomAttributes.Add(attribute.Key, attribute.Value);
 
             foreach (var node in Nodes)
-                newObj.Nodes.Add(node is SvgElement ? newObj.Children[Children.IndexOf((SvgElement)node)] : node.DeepCopy());
+            {
+                if (node is SvgElement)
+                {
+                    var index = Children.IndexOf((SvgElement)node);
+                    if (index >= 0)
+                    {
+                        newObj.Nodes.Add(newObj.Children[index]);
+                        continue;
+                    }
+                }
+                newObj.Nodes.Add(node.DeepCopy());
+            }

```

Comment:
Is it guaranteed that at this point the elements have already been copied? Otherwise, we would need two passes, where first the objects are collected and copied in the second pass if not already copied.

Reply:
`DeepCopy` of `Children` is done just before.(line 986)

Reply:
Ah, right of course. Thanks!